cmd/prometheus/main.go

```diff
@@ -85,12 +85,15 @@ func Main() int {
 		log.Errorf("Error initializing remote storage: %s", err)
 		return 1
 	}
-
 	if remoteStorage != nil {
 		sampleAppender = append(sampleAppender, remoteStorage)
 		reloadables = append(reloadables, remoteStorage)
 	}
 
+	configurableRemoteStorage := remote.NewConfigurable()
+	sampleAppender = append(sampleAppender, configurableRemoteStorage)
+	reloadables = append(reloadables, configurableRemoteStorage)
```

Comment:
I think we're missing a shutdown for the configurable remote storage at Prometheus shutdown, similar to https://github.com/tomwilkie/prometheus/blob/3db53d4f30ce554a8dbb784225781469f4aa97f2/cmd/prometheus/main.go#L174